src/pages/workspaces/workspace/ShareWorkspaceModal.js

```diff
@@ -253,13 +261,52 @@ const ShareWorkspaceModal = ({ onDismiss, workspace, workspace: { workspace: { n
     searchValueValid && !searchHasFocus && p(addUserReminder),
     h2({ style: { ...Style.elements.sectionHeader, margin: '1rem 0 0.5rem 0' } }, ['Current Collaborators']),
     div({ ref: list, role: 'list', style: styles.currentCollaboratorsArea }, [
-      h(Fragment, _.map(renderCollaborator, Utils.toIndexPairs(acl))),
+      h(Fragment, _.flow(
+        _.remove(aclEntryIsTerraSupport),
+        Utils.toIndexPairs,
+        _.map(renderCollaborator)
+      )(acl)),
       !loaded && centeredSpinner()
     ]),
     updateError && div({ style: { marginTop: '1rem' } }, [
       div(['An error occurred:']),
       updateError
     ]),
+    div({ style: { ...modalStyles.buttonRow, justifyContent: 'space-between' } }, [
+      h(IdContainer, [
+        id => h(TooltipTrigger, {
+          content: terraSupportAccess ?
+            `Terra Support has ${_.toLower(terraSupportAccess.accessLevel)} access to this workspace` :
```

Comment:
I found the tooltip a bit confusing when granting access. When I toggle the switch to True, the tooltip immediately says that Terra Support has reader access. I was confused about whether I needed to press Save, or if the operation truly happened when the switch was changed (I experimented to confirm that I actually do need to press Save).

I think it would make more sense for the tooltip to reflect the actual state, or have a different message for this intermediary state (perhaps "Saving will grant Terra Support reader access….", or "Terra Support will have reader access…").

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/2902/commits/e83afd6027c1654a15850e0ad8288ad8e9859947.